server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -249,34 +313,50 @@ export class TenantManager {
 
         const encryptedTenantKey1 = tenantDocument.key;
         const tenantKey1 = this.secretManager.decryptSecret(encryptedTenantKey1);
+        let tenantKey2;
+
         if (tenantKey1 == null) {
             winston.error("Tenant key1 decryption failed.");
-            Lumberjack.error("Tenant key1 decryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
+            Lumberjack.error("Tenant key1 decryption failed.", lumberProperties);
             throw new NetworkError(500, "Tenant key1 decryption failed.");
         }
 
         const encryptedTenantKey2 = tenantDocument.secondaryKey;
-        if (!encryptedTenantKey2) {
+        if (encryptedTenantKey2) {
+            tenantKey2 = this.secretManager.decryptSecret(encryptedTenantKey2);
+            if (tenantKey2 == null) {
+                winston.error("Tenant key2 decryption failed");
+                Lumberjack.error("Tenant key2 decryption failed.", lumberProperties);
+                throw new NetworkError(500, "Tenant key2 decryption failed.");
+            }
+        } else {
```

Comment:
I know @zhangxin511 recommended refactoring this to invert the logic, but personally I think that this nested if logic made things more confusing and error-prone. Earlier, we had earlier returns, allowing to not have nested conditionals. Coding style is definitely something subjective, but perhaps we should ask the person who originally wrote this piece of code (@PingZhu2232 ) to provide some feedback and, if we decide to maintain this new refactored logic, validate it works properly?

Reply:
Personally I prefer guard clause, but in the code block here it doesn't have too many nested if clause, either way should work. Agreed on validation before pushing to prod.

Reply:
How about this?
``` 
        const encryptedTenantKey2 = tenantDocument.secondaryKey;
        const tenantKey2 = encryptedTenantKey2 ? this.secretManager.decryptSecret(encryptedTenantKey2) || "";         if (tenantKey2 == null) {
               winston.error("Tenant key2 decryption failed");
               Lumberjack.error("Tenant key2 decryption failed.", lumberProperties);
               throw new NetworkError(500, "Tenant key2 decryption failed.");
        } 
       if (encryptedTenantKey2 ) { // can we also do || tenantKey2 === "" here?
            winston.info("Tenant key2 doesn't exist.");
            Lumberjack.info("Tenant key2 doesn't exist.", lumberProperties);
       }
        const cacheKeys = {
            key1: encryptedTenantKey1,
            key2: encryptedTenantKey2,
        };
        await this.setCacheKey(tenantId, cacheKeys, fetchTenantKeyMetric); 
       return {
            key1: tenantKey1,
            key2: tenantKey2,
        };

````

Reply:
@zhangxin511 and I discussed this I tested the proposed change. Everything looks like it is working ok

Reply:
@hedasilv @PingZhu2232 What do you think about this change? It removes the nested if clause and I think it is clearer

Reply:
Sounds good! Thanks for validating it thorougly.